src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -475,6 +475,32 @@ void QgsMssqlProvider::loadFields()
   }
 }
 
+QString QgsMssqlProvider::quotedValue( const QVariant& value )
+{
+  if ( value.isNull() )
+    return "NULL";
+
+  switch ( value.type() )
+  {
+    case QVariant::Int:
+    case QVariant::LongLong:
+    case QVariant::Double:
+      return value.toString();
+
+    case QVariant::Bool:
+      return value.toBool() ? "1" : "0";
```

Comment:
Does this work ok? The mssql expression compiler uses a "1=1"/"1=0" hack around the lack of boolean literals. It'd be nice to remove the duplicate quoted value function and use this one for the expression compiler too